alf/utils/value_ops.py

```diff
@@ -187,18 +180,14 @@ def one_step_discounted_return(rewards, values, step_types, discounts):
         A tensor with shape [T-1, B] (or [T-1]) representing the discounted
         returns.
     """
-    assert values.shape[0] >= 2, "The sequence length needs to be \
-                                at least 2. Got {s}".format(s=values.shape[0])
-
-    discounts = discounts[1:]
-    rewards = rewards[1:]
-    values = values[1:]
-    step_types = step_types[:-1]
+    assert values.shape[0] >= 2, ("The sequence length needs to be "
+                                  "at least 2. Got {s}".format(
+                                      s=values.shape[0]))
 
     is_lasts = (step_types == StepType.LAST).to(dtype=torch.float32)
-    returns = rewards + (1 - is_lasts) * discounts * values
+    rets = rewards[1:] + (1 - is_lasts[:-1]) * discounts[1:] * values[1:]
```

Comment:
It seems that `rets = (rewards + discounts * values)[1:]` should be fine with the understanding the the loss at StepType.LAST will be ignored. And the rets from LINE 188 at StepType.LAST is not correct either.

Reply:
Updated for better understanding